content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2416 @@
+is a notable deviation from the treatment of conventional simple numeric
+samples, which are always treated as floats throughout the whole stack for the
+sake of simplicity.
+
+The main reason for the more involved treatment of histograms is the easy
+efficiency gains in protobuf-based exposition formats. Protobuf uses varint
+encoding for integers, which reduces the data size for small integer values
+without requiring an additional compression layer. This benefit is amplified by
+the [delta encoding of integer buckets](#buckets), which generally results in
+smaller integer values. Floats, in contrast, always require 8 bytes in
+protobuf. In practice, many integers in an integer histogram will fit in 1
+byte, and most will fit in 2 bytes, so that the explicit presence of integer
+histogram in a protobuf-exposition format results directly in a data size
+reduction approaching 8x for histograms with many buckets. This is particularly
+relevant as the overwhelming majority of histograms exposed by instrumented
+targets are integer histograms.
+
+For similar reasons, the representation of integer histograms in RAM and on
+disk is generally more efficient than that of float histograms. This is less
+relevant than the benefits in the exposition format, though. For one,
+Prometheus uses Gorilla-style XOR encoding for floats, which reduces their
+size, albeit not as much as the double-delta encoding used for integers. More
+importantly, an implementation could always decide to internally use an integer
+representation for histogram fields that are effectively integer values (see
+above). (Historical note: Prometheus v1 used exactly this approach to improve
+the compression of float samples, and Prometheus v3 might very well adopt this
+approach again in the future.)
+
+In a counter histogram, the total _count_ of observation and the counts in the
+buckets individually behave like Prometheus counters, i.e. they only go down
+upon a counter reset. However, the _sum_ of observation may decrease as a
+consequence of the observation of negative values. PromQL implementations MUST
+detect counter resets based on the whole histogram (see the [counter reset
+considerations section](#counter-reset-considerations) below for details).
+(Note that this always has been a problem for the _sum_ component of classic
+histograms and summaries, too. The approach so far was to accept that counter
+reset detection silently breaks for _sum_ in those cases. Fortunately, negative
+observations are a very rare use case for Prometheus histograms and summaries.)
+
+### Schema
+
+The _schema_ is a signed integer value with a size of 8 bits (short: int8). It
+defines the way bucket boundaries are calculated. The currently valid values
+are -53 and the range between and including -4 and +8. More schemas may be
+added in the future. -53 is a schema for so-called _custom bucket boundaries_
+or short _custom buckets_, while the other schema numbers represent the
```

Comment:
Let's add those names to glossary 🤗

Reply:
Maybe we can consider extending the glossary to all the new terms introduced by this document. However, there would be a lot. If this was a printed document, it might still be helpful for cross reference, but since this is electronic, the reader will simply search for any term they don't remember. (The few terms in the glossary so far are those that might easily be misunderstood or confused.)